icu4j/main/tests/core/src/com/ibm/icu/dev/test/message2/FromJsonTest.java

```diff
@@ -0,0 +1,412 @@
+// © 2022 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+package com.ibm.icu.dev.test.message2;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import com.ibm.icu.dev.test.TestFmwk;
+
+@RunWith(JUnit4.class)
+@SuppressWarnings("javadoc")
+public class FromJsonTest extends TestFmwk {
+
+    // These tests come from the test suite created for the JavaScript implementation of MessageFormat2:
+    // https://github.com/messageformat/messageformat/blob/master/packages/mf2-messageformat/src/__fixtures/test-messages.json
+    static final TestCase[] TEST_CASES = {
+            new TestCase.Builder()
+                .pattern("{hello}")
+                .expected("hello")
+                .build(),
+            new TestCase.Builder()
+                .pattern("{hello {(world)}}")
+                .expected("hello world")
+                .build(),
+            new TestCase.Builder()
+                .pattern("{hello {()}}")
+                .expected("hello ")
+                .build(),
+            new TestCase.Builder()
+                .pattern("{hello {$place}}")
+                .arguments(Args.of( "place", "world" ))
+                .expected("hello world")
+                .build(),
+            new TestCase.Builder()
+                .pattern("{hello {$place}}")
+                .expected("hello {$place}")
+                // errorsJs: ["missing-var"]
+                .build(),
+            new TestCase.Builder()
+                .pattern("{{$one} and {$two}}")
+                .arguments(Args.of( "one", 1.3, "two", 4.2 ))
+                .expected("1.3 and 4.2")
+                .build(),
+            new TestCase.Builder()
+                .pattern("{{$one} et {$two}}")
+                .locale("fr")
+                .arguments(Args.of( "one", 1.3, "two", 4.2 ))
+                .expected("1,3 et 4,2")
+                .build(),
+            new TestCase.Builder()
+                .pattern("{hello {(4.2) :number}}")
+                .expected("hello 4.2")
```

Comment:
🎉 !
... and looking forward to seeing this with locale "ar-EG" :-)

Reply:
Added it, just for fun :-)